src/Polly.Shared/Timeout/TimeoutEngine.cs

```diff
@@ -47,8 +47,14 @@ internal static partial class TimeoutEngine
                         .Run(() =>
                             action(context, combinedToken)       // cancellation token here allows the user delegate to react to cancellation: possibly clear up; then throw an OperationCanceledException.
                             , combinedToken);           // cancellation token here only allows Task.Run() to not begin the passed delegate at all, if cancellation occurs prior to invoking the delegate.  
-
-                        actionTask.Wait(timeoutCancellationTokenSource.Token); // cancellation token here cancels the Wait() and causes it to throw, but does not cancel actionTask.  We use only timeoutCancellationTokenSource.Token here, not combinedToken.  If we allowed the user's cancellation token to cancel the Wait(), in this pessimistic scenario where the user delegate may not observe that cancellation, that would create a no-longer-observed task.  That task could in turn later fault before completing, risking an UnobservedTaskException.
+                        try
+                        {
+                            actionTask.Wait(timeoutCancellationTokenSource.Token); // cancellation token here cancels the Wait() and causes it to throw, but does not cancel actionTask.  We use only timeoutCancellationTokenSource.Token here, not combinedToken.  If we allowed the user's cancellation token to cancel the Wait(), in this pessimistic scenario where the user delegate may not observe that cancellation, that would create a no-longer-observed task.  That task could in turn later fault before completing, risking an UnobservedTaskException.
+                        }
+                        catch (AggregateException ex) when (ex.InnerExceptions.Count == 1)
```

Comment:
Thanks @udlose for the observations!  Re:

>And what do you think should happen when there happen to be multiple inner exceptions? 

A general `Task` could, as you say @udlose , throw an `AggregateException` with either single or multiple `InnerExceptions`.  In this case however we know [the specific way this `Task` is invoked](https://github.com/hambudi/Polly/blob/fixAggregateExceptionsInPessimisticsTimeout/src/Polly.Shared/Timeout/TimeoutEngine.cs#L47-L48).  In all tests empirically, this is throwing an `AggregateException` with a single `InnerException`.  Even if the `myDelegate` executed as `TimeoutPolicy.Execute(() => myDelegate)` throws multiple inner exceptions, the result is an: 
`AggregateException -> AggregateException -> (multiple inner exceptions)`, and we can (as the PR does) strip off the outer `AggregateException`, so that the exception the policy rethrows matches the exception that would be thrown if the delegate was not executed through pessimistic `TimeoutPolicy`  - this is the goal.

Given our confidence that all `AggregateException`s thrown here will be an `AggregateException` with single `InnerException`, we could plausibly drop the [`ex.InnerExceptions.Count == 1` check](https://github.com/hambudi/Polly/blob/fixAggregateExceptionsInPessimisticsTimeout/src/Polly.Shared/Timeout/TimeoutEngine.cs#L54).  However, I suggest we retain it, as it prevents Polly ever _invisibly throwing away_ information, should the delegate throw an aggregate-with-mutliple-inners exception.

Re:

>Something like ex.Flatten() should probably be used here.

Couple of reasons why we have chosen not to `ex.Flatten()` within Polly.  First, a principle for Polly policies is that they should not cause any side effects to the way the user's delegate is executed, or expresses exceptions, beyond the stated aim of the policy.  If we choose to `ex.Flatten()` within the library, that causes a side effect as it _changes_ the way exceptions from the user's delegate are expressed.  Second, if we choose to `ex.Flatten()` from within the library, we _take away from the user the choice_ of whether to `ex.Flatten()`.